contracts/contracts/core/vault/src/main.sw

```diff
@@ -1187,355 +745,151 @@ fn _get_after_fee_amount(
+    let asset_price = pricefeed_wrapper.price(asset);
+    // TODO overflow check
+    asset_price * PRICE_PRECISION / collateral_asset_price
 }
 
 #[storage(read)]
-fn _get_min_price(asset: AssetId) -> u256 {
-    let vault_pricefeed = abi(VaultPricefeed, storage.pricefeed_provider.read().into());
-    vault_pricefeed.get_price(
-        asset, 
-        false,
-    )
+fn _get_min_price(asset: b256) -> u256 {
+    let pricefeed_wrapper = abi(PricefeedWrapper, PRICEFEED_WRAPPER.bits());
+    let collateral_asset_price = pricefeed_wrapper.price(COLLATERAL_ASSET);
+    let asset_price = pricefeed_wrapper.price(asset);
+    // TODO overflow check
+    asset_price * PRICE_PRECISION / collateral_asset_price
 }
 
 #[storage(read)]
-fn _is_asset_whitelisted(asset: AssetId) -> bool {
-    storage.whitelisted_assets.get(asset).try_read().unwrap_or(false)
+fn _is_asset_whitelisted(asset: b256) -> bool {
+    storage::vault.whitelisted_assets.get(asset).try_read().unwrap_or(false)
 }
 
 #[storage(read)]
 fn _get_position_by_key(position_key: b256) -> Position {
-    storage.positions
+    storage::vault.positions
         .get(position_key).try_read().unwrap_or(Position::default())
 }
 
 #[storage(read)]
-fn _get_pool_amounts(asset: AssetId) -> u256 {
-    storage.pool_amounts.get(asset).try_read().unwrap_or(0)
-}
-
-#[storage(read)]
-fn _get_buffer_amounts(asset: AssetId) -> u256 {
-    storage.buffer_amounts.get(asset).try_read().unwrap_or(0)
-}
-
-#[storage(read)]
-fn _get_fee_reserves(asset: AssetId) -> u256 {
-    storage.fee_reserves.get(asset).try_read().unwrap_or(0)
-}
-
-#[storage(read)]
-fn _get_reserved_amount(asset: AssetId) -> u256 {
-    storage.reserved_amounts.get(asset).try_read().unwrap_or(0)
-}
-
-#[storage(read)]
-fn _get_global_short_sizes(asset: AssetId) -> u256 {
-    storage.global_short_sizes.get(asset).try_read().unwrap_or(0)
-}
-
-#[storage(read)]
-fn _get_rusd_amount(asset: AssetId) -> u256 {
-    storage.rusd_amounts.get(asset).try_read().unwrap_or(0)
-}
-
-#[storage(read)]
-fn _get_guaranteed_usd(asset: AssetId) -> u256 {
-    storage.guaranteed_usd.get(asset).try_read().unwrap_or(0)
+fn _get_pool_amounts(asset: b256) -> u256 {
+    storage::vault.pool_amounts.get(asset).try_read().unwrap_or(0)
 }
 
 #[storage(read)]
-fn _get_cumulative_funding_rate(asset: AssetId) -> u256 {
-    storage.cumulative_funding_rates.get(asset).try_read().unwrap_or(0)
+fn _get_fee_reserve() -> u256 {
+    storage::vault.fee_reserve.try_read().unwrap_or(0)
 }
 
 fn _get_position_key(
     account: Identity,
-    collateral_asset: AssetId,
-    index_asset: AssetId,
+    index_asset: b256,
     is_long: bool,
 ) -> b256 {
     keccak256(PositionKey {
         account,
-        collateral_asset,
         index_asset,
         is_long,
     })
 }
 
-// note that if calling this function independently the cumulativeFundingRates 
-// used in getFundingFee will not be the latest value
 #[storage(read)]
 fn _validate_liquidation(
```

Comment:
this returns some flags as first element of the returned tuple. I believe that should be a descriptive enum.

Reply:
There is another PR that modifies liquidation function, I will provide a fix there.